v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+
+typedef struct PARSED_EVENT_TAG
+{
+    char event_name[MAX_EVENT_NAME_LENGTH];
+    char content[MAX_CONTENT_LENGTH];
+    char file[MAX_FILE_LENGTH];
+    char func[MAX_FUNC_LENGTH];
+    int line;
+    uint32_t property_count;
+    PARSED_EVENT_PROPERTY properties[MAX_PROPERTY_COUNT];
+} PARSED_EVENT;
+
+#define MAX_EVENTS 128
+
+static volatile LONG parsed_event_count;
+static PARSED_EVENT parsed_events[MAX_EVENTS];
+static HANDLE process_thread_handle;
+static TRACEHANDLE trace;
+static char logger_name[128];
+static char trace_session_name[128];
+
+// This callback is needed
+// Without this callback events are not properly processed
+static ULONG WINAPI event_trace_buffer_callback(PEVENT_TRACE_LOGFILEA log_file)
+{
+    (void)log_file;
+
+    return TRUE;
+}
+
+static void WINAPI event_trace_record_callback(EVENT_RECORD* pEventRecord)
+{
+    if (IsEqualGUID(&pEventRecord->EventHeader.ProviderId, &provider_guid))
+    {
+        const uint8_t* metadata = NULL;
+
+        LONG current_parsed_event_count = InterlockedAdd(&parsed_event_count, 0);
+        (void)memset(&parsed_events[current_parsed_event_count], 0, sizeof(PARSED_EVENT));
+
+        EVENT_HEADER_EXTENDED_DATA_ITEM* extended_data_items = pEventRecord->ExtendedData;
+
+        for (unsigned short i = 0; i < pEventRecord->ExtendedDataCount; i++)
+        {
+            if (extended_data_items[i].ExtType == EVENT_HEADER_EXT_TYPE_EVENT_SCHEMA_TL)
+            {
+                // This one has the event name
+                // Not really documented, but the Windows source code seems to indicate I can just skip over a few bytes that hold a length and version
+                const char* event_name = (const char*)(uintptr_t)(extended_data_items[i].DataPtr + 4);
+
+                if (event_name != NULL)
+                {
+                    (void)strncpy(parsed_events[current_parsed_event_count].event_name, event_name, sizeof(parsed_events[current_parsed_event_count].event_name));
+                    parsed_events[current_parsed_event_count].event_name[sizeof(parsed_events[current_parsed_event_count].event_name) - 1] = '\0';
+                    metadata = (const uint8_t*)(event_name + strlen(event_name) + 1);
+                }
+
+                break;
+            }
+        }
+       
+        POOR_MANS_ASSERT(pEventRecord->UserData != NULL);
+
+        const char* current_user_data_field = pEventRecord->UserData;
+        if (current_user_data_field != NULL)
+        {
+            (void)strncpy(parsed_events[current_parsed_event_count].content, current_user_data_field, sizeof(parsed_events[current_parsed_event_count].content));
+            parsed_events[current_parsed_event_count].content[sizeof(parsed_events[current_parsed_event_count].content) - 1] = '\0';
+            current_user_data_field += strlen(current_user_data_field) + 1;
+        }
+        POOR_MANS_ASSERT(strcmp((const char*)metadata, "content") == 0);
+        metadata += sizeof("content") + 1;
```

Comment:
>+ 1 [](http://example.com/codeflow?start=38&length=3)

rarely do I see sizeof("some string") + 1. It is usually " -1 " to get the length in characters. What's this +1 byte that needs skipping? A comment would be nice for the reviewer :)

Reply:
Added comment.